src/App.js

```diff
@@ -14,11 +15,69 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selected: [],
+  };
+
+  clear = () => {
+    this.setState({ selected: [] });
+  };
+
+  toggle = (product) => {
+    const { selected } = this.state;
+
+    selected.includes(product)
+      ? this.setState({
+        selected: selected.filter(goods => goods !== product),
+      })
+      : this.setState({ selected: [...selected, product] });
+  };
+
+  render() {
+    const { selected } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          {'Selected goods: '}
+          {<span>{selected.join(', ')}</span>}
+        </h1>
+
+        {selected.length ? (
+          <button
+            type="button"
+            className="button"
+            onClick={this.clear}
+          >
+            Clear
+          </button>
+        )
+          : null}
+
+        <ul className="products">
+          {goodsFromServer.map(product => (
+            <li
+              key={product}
+              className={classNames('products__item', {
+                active: selected.includes(product),
+              })}
+            >
+              <span>{product}</span>
+
+              <button
+                onClick={() => this.toggle(product)}
+                className="button products__button"
+                type="button"
+              >
+                +
```

Comment:
```suggestion
                {selected.includes(product) ? '-' : '+'}
```

Just FYI. In such a way you can swap +/- depending on if good is selected or not